File: android/src/main/java/com/owliverse/yt_player/YtPlayerViewActivity.java
package com.owliverse.yt_player;

import android.content.Intent;
import android.os.Bundle;

import com.google.android.youtube.player.YouTubePlayer;
import com.google.android.youtube.player.YouTubePlayerView;

public class YtPlayerViewActivity extends YouTubeFailureRecoveryActivity {

    private String apiKey;
    private String videoId;
    private boolean autoPlay;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_yt_player_view);

        final Intent intent = getIntent();
        apiKey = intent.getStringExtra("apiKey");
        videoId = intent.getStringExtra("videoId");
        autoPlay = intent.getBooleanExtra("autoPlay", false);
        YouTubePlayerView youTubeView = (YouTubePlayerView) findViewById(R.id.youtube_view);
        youTubeView.initialize(getApiKey(), this);
    }

    @Override
    protected YouTubePlayer.Provider getYouTubePlayerProvider() {
        return (YouTubePlayerView) findViewById(R.id.youtube_view);
    }

    @Override
    protected String getApiKey() {
        return apiKey;
    }

    @Override
    public void onInitializationSuccess(YouTubePlayer.Provider provider, YouTubePlayer youTubePlayer, boolean wasRestored) {
        if (!wasRestored) {
            youTubePlayer.setFullscreen(true);
            youTubePlayer.setShowFullscreenButton(false);
            if (autoPlay) {
                youTubePlayer.loadVideo(videoId);
            } else {
                youTubePlayer.cueVideo(videoId);
            }
        }
    }
}

File: android/src/main/kotlin/com/owliverse/yt_player/YtPlayerViewActivity.kt
package com.owliverse.yt_player

import android.content.Context
import android.content.Intent
import com.google.android.youtube.player.YouTubeApiServiceUtil
import com.google.android.youtube.player.YouTubeInitializationResult
import com.owliverse.yt_player.YtPlayerViewActivity
import com.google.android.youtube.player.YouTubeIntents
import com.owliverse.yt_player.YtPlayerWebViewActivity
import com.owliverse.yt_player.YtPlayerPlugin
import com.owliverse.yt_player.YouTubeFailureRecoveryActivity
import android.os.Bundle
import com.owliverse.yt_player.R
import com.google.android.youtube.player.YouTubePlayerView
import android.view.View
import com.google.android.youtube.player.YouTubePlayer
import androidx.appcompat.app.AppCompatActivity
import android.view.KeyEvent
import com.google.android.youtube.player.YouTubeBaseActivity
import android.widget.Toast

class YtPlayerViewActivity : YouTubeFailureRecoveryActivity() {

    private lateinit var videoId: String
    private var autoPlay = false

    override lateinit var apiKey: String
    override lateinit var youTubePlayerProvider: YouTubePlayer.Provider

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_yt_player_view)
        apiKey = intent.getStringExtra("apiKey")!!
        videoId = intent.getStringExtra("videoId")!!
        autoPlay = intent.getBooleanExtra("autoPlay", false)
        findViewById<YouTubePlayerView>(R.id.youtube_view).let {
            youTubePlayerProvider = it
            it.initialize(apiKey, this)
        }
    }

    override fun onInitializationSuccess(
        provider: YouTubePlayer.Provider,
        youTubePlayer: YouTubePlayer,
        wasRestored: Boolean
    ) {
        if (!wasRestored) {
            youTubePlayer.setFullscreen(true)
            youTubePlayer.setShowFullscreenButton(false)
            if (autoPlay) {
                youTubePlayer.loadVideo(videoId)
            } else {
                youTubePlayer.cueVideo(videoId)
            }
        }
    }
}